dask/dataframe/tests/test_shuffle.py

```diff
@@ -1459,3 +1462,14 @@ def test_sort_values_bool_ascending():
     # attempt to sort with list of ascending booleans
     with pytest.raises(NotImplementedError):
         ddf.sort_values(by="a", ascending=[True, False])
+
+
+def test_set_index_no_resursion_error(client):
+    # see: https://github.com/dask/dask/issues/8955
+    ddf = (
+        dask.datasets.timeseries(end="2000-07-01")
+        .reset_index()
+        .astype({"timestamp": str})
+    )
+    ddf = ddf.set_index("timestamp", sorted=True)
+    ddf.compute()
```

Comment:
It's very rare to use distributed in the Dask test suite.  We like to keep those systems separate when possible.

Can you see if this failure triggers if you set `with dask.config.set(shuffle="tasks"):` around everything and leave distributed out of it?  That's the main change with shuffling that occurs.  The local scheduler and the distributed scheduler use separate shuffle algorithms.

Reply:
Also, just computing this doesn't sound like a very precise test.  Was there something bad that was happening before that we should be calling out in particular?  I know that the problem presented as "things were really slow" but why was that?  Was it because we weren't respecting the sorted index?  Was it because of something else?  It would be nice to assert that whatever was broken before is clean now, if that's easy (if it't not easy then please ignore)

Reply:
Hey Matt -- thanks for the feedback. It's good to know to avoid using distributed directly. I'll try the configuration context manager.

What I'm testing for here is a RecursionError (which is triggered by this test at least on my laptop). I initially tried wrapping the meat of the test with:

```python
with pytest.raises(RecursionError):
    ...
```

But that wasn't catching it and the console was getting clobbered with >300 lines of traceback.

I had to run to an appointment yesterday afternoon, so I pushed this just to make it visible that I was working on it.

Reply:
Hey Matt -- the `dask.config` context manager doesn't trigger the RecursionError the same way that the `client` fixture from distributed does.

More broadly, the goal of this test is (now that Julia's commits are in the PR) to confirm that the reproducer in the original issues *doesn't* trigger a RecursionError.

Naively, simply calling `.compute()` _could_ achieve that. But if you have pointer to a more elegant and/or robust approach, I'm all ears.

Reply:
> Hey Matt -- the dask.config context manager doesn't trigger the RecursionError the same way that the client fixture from distributed does.

Well drat.